cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java

```diff
@@ -1367,16 +1382,150 @@ private static void writeDCNode(String id, String dcNodeName, String dcEquipment
         DCNodeEq.write(id, dcNodeName, dcEquipmentContainerId, cimNamespace, writer, context);
     }
 
-    private static void writeDCGround(String id, String dcGroundName, String cimNamespace, XMLStreamWriter writer, CgmesExportContext context) throws XMLStreamException {
-        DCGroundEq.write(id, dcGroundName, cimNamespace, writer, context);
+    private static void writeDCGround(String id, String dcGroundName, double resistance, String cimNamespace, XMLStreamWriter writer, CgmesExportContext context) throws XMLStreamException {
+        DCGroundEq.write(id, dcGroundName, resistance, cimNamespace, writer, context);
+    }
+
+    private static void writeDcTerminal(DcTerminal dcTerminal, String cimNamespace, XMLStreamWriter writer, CgmesExportContext context) throws XMLStreamException {
+        int sequenceNumber;
+        DcConnectable<?> dcConnectable = dcTerminal.getDcConnectable();
+        if (dcConnectable instanceof DcGround) {
+            sequenceNumber = 1;
+        } else if (dcConnectable instanceof AcDcConverter) {
+            sequenceNumber = dcTerminal.getTerminalNumber().getNum();
+        } else {
+            sequenceNumber = dcTerminal.getSide().getNum();
+        }
+        writeDcTerminal(dcTerminal.getDcConnectable(), dcTerminal.getDcNode(), sequenceNumber, cimNamespace, writer, context);
+    }
+
+    private static void writeDcTerminal(Identifiable<?> dcIdentifiable, DcNode dcNode, int sequenceNumber, String cimNamespace, XMLStreamWriter writer, CgmesExportContext context) throws XMLStreamException {
+        String dcIdentifiableId = context.getNamingStrategy().getCgmesId(dcIdentifiable);
+        String dcNodeId = null;
+        if (!context.isBusBranchExport()) {
+            dcNodeId = context.getNamingStrategy().getCgmesId(dcNode);
+        } else if (!context.isCim16BusBranchExport()) {
+            dcNodeId = context.getNamingStrategy().getCgmesId(dcNode.getDcBus());
+        }
```

Comment:
Just to make it easier to read:

```suggestion
        if (!context.isBusBranchExport()) {
            // node-breaker export
            dcNodeId = context.getNamingStrategy().getCgmesId(dcNode);
        } else if (!context.isCim16BusBranchExport()) {
            // CIM100 bus-branch export
            dcNodeId = context.getNamingStrategy().getCgmesId(dcNode.getDcBus());
        }
```